### PR TITLE
fixes for build issues

### PR DESCRIPTION
**Pull request by sdeoras**
this pr fixes several lint and errcheck issues associated with the build. Pl. note that errcheck does not pass on coreos repos, so the only choice i had was to not do err check on it. Accordingly errcheck was removed from etcd v2 and v3.

### CODE REVIEW

File: etcd/common/common.go [hunk 1]
```diff
@@ -166,46 +169,47 @@ func Version(url string, options map[string]string) (string, error) {
...
 		// Probably a version less than 2.3. Default to using v2 apis
 		return kvdb.EtcdBaseVersion, nil
 	}
-	if version.Server == "" {
+	if ver.Server == "" {
 		// This should never happen in an ideal scenario unless
 		// etcd messes up. To avoid a crash further in this code
 		// we return an error
-		return "", fmt.Errorf("Unable to determine etcd version (empty response from etcd)")
+		return "", fmt.Errorf("unable to determine etcd version (empty response from etcd)")
```
[the remark this answers is not in the document]
Comment (sdeoras) [reply etcd/common/common.go]: generally recommended to start with lowercase: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: common/common.go [hunk 2]
```diff
@@ -38,19 +38,22 @@ type BaseKvdb struct {
 	lock sync.Mutex
 }
 
+// SetFatalCb sets fatal func callback.
```
Comment (jvinod) [inline common/common.go]: SetFatal callback is invoked when an unrecoverable KVDB error happens.

File: common/common.go [hunk 3]
```diff
@@ -38,19 +38,22 @@ type BaseKvdb struct {
...
 func (b *BaseKvdb) SetFatalCb(f kvdb.FatalErrorCB) {
 	b.lock.Lock()
 	defer b.lock.Unlock()
 	b.FatalCb = f
 }
 
+// SetLockTimeout sets lock timeout.
```
Comment (jvinod) [inline common/common.go]: If the lock is held past this duration, then a configured fatal callback is called.

File: common/common.go [hunk 4]
```diff
@@ -63,22 +66,26 @@ func (b *BaseKvdb) CheckLockTimeout(
...
 }
 
+// LockTimedout does lock timedout.
 func (b *BaseKvdb) LockTimedout(key string) {
 	b.lock.Lock()
 	defer b.lock.Unlock()
 	b.lockTimedout(key)
 }
 
+// lockTimedout does lock timedout.
```
Comment (jvinod) [inline common/common.go]: Function is invoked if lock is held past configured timeout.

File: common/common.go [hunk 5]
```diff
@@ -63,22 +66,26 @@ func (b *BaseKvdb) CheckLockTimeout(
...
 func (b *BaseKvdb) lockTimedout(key string) {
 	b.FatalCb("Lock %s hold timeout triggered", key)
 }
 
+// SerializeAll serializes all.
```
Comment (jvinod) [inline common/common.go]: Serializes all key value pairs to a byte array.

File: common/common.go [hunk 6]
```diff
@@ -87,6 +94,7 @@ func (b *BaseKvdb) SerializeAll(kvps kvdb.KVPairs) ([]byte, error) {
 	return out, nil
 }
 
+// DeserializeAll deserializes all.
```
Comment (jvinod) [inline common/common.go]: Unmarshals a byte stream created from serializeAll into the kvdb tree.

File: consul/kv_consul.go [hunk 7]
```diff
@@ -271,8 +271,10 @@ func (kv *consulKV) Create(
 			if ok && err == nil {
 				return kvPair, err
 			}
-			kv.client.Session().Destroy(sessionPair.Session, nil)
-			kv.Delete(key)
+			_, _ = kv.client.Session().Destroy(sessionPair.Session, nil)
+			if _, err := kv.Delete(key); err != nil {
+				return nil, err
```
Comment (jvinod) [inline consul/kv_consul.go]: I think you still want to return the error from `kv.client.KV().Acquire(`

**Pull request merged**